include/tcframe/io_manipulator/LinesIOSegmentManipulator.hpp

```diff
@@ -26,24 +26,29 @@ class LinesIOSegmentManipulator {
             }
         }
 
+        string lastVariableName;
+
         for (int j = 0; j < *segment->size(); j++) {
-            string lastVariableName;
+            string lastVariableNameInSegment;
             for (Variable* variable : segment->variables()) {
                 if (variable->type() == VariableType::VECTOR) {
-                    if (!lastVariableName.empty()) {
-                        WhitespaceManipulator::parseSpace(in, lastVariableName);
+                    if (!lastVariableNameInSegment.empty()) {
+                        WhitespaceManipulator::parseSpace(in, lastVariableNameInSegment);
                     }
                     ((Vector*) variable)->parseAndAddElementFrom(in);
                 } else {
-                    if (!lastVariableName.empty() && !WhitespaceManipulator::canParseNewline(in)) {
-                        WhitespaceManipulator::parseSpace(in, lastVariableName);
+                    if (!lastVariableNameInSegment.empty() && !WhitespaceManipulator::canParseNewline(in)) {
+                        WhitespaceManipulator::parseSpace(in, lastVariableNameInSegment);
                     }
                     ((Matrix*) variable)->parseAndAddRowFrom(in, j);
                 }
-                lastVariableName = TokenFormatter::formatVectorElement(variable->name(), j);
+                lastVariableNameInSegment = TokenFormatter::formatVectorElement(variable->name(), j);
             }
-            WhitespaceManipulator::parseNewline(in, lastVariableName);
+            WhitespaceManipulator::parseNewline(in, lastVariableNameInSegment);
+            lastVariableName = lastVariableNameInSegment;
```

Comment:
Done. Previously I still use it to check whether it is currently reading the first column (to know whether need to read space), but now I used bool instead.